tst/EnergyPlus/unit/OutputReportTabular.unit.cc

```diff
@@ -9610,3 +9611,98 @@ TEST_F(SQLiteFixture, ORT_EndUseBySubcategorySQL_DualUnits)
         ASSERT_EQ(13u, result.size()) << "Failed for query: " << query;
     }
 }
+
+
+TEST_F(SQLiteFixture, OutputReportTabularTest_EscapeHTML)
+{
+    // Test for #8542 - Ensures strings are escaped before going to HTML
+    EnergyPlus::sqlite->sqliteBegin();
+    EnergyPlus::sqlite->createSQLiteSimulationsRecord(1, "EnergyPlus Version", "Current Time");
+
+    auto &ort(state->dataOutRptTab);
+    ort->numStyles = 1;
+    ort->TableStyle(1) = OutputReportTabular::iTableStyle::HTML;
+    ort->del(1) = DataStringGlobals::CharSpace; // space - this is not used much for HTML output
+
+    ort->WriteTabularFiles = true;
+
+    SetupUnitConversions(*state);
+    ort->unitsStyle = OutputReportTabular::iUnitsStyle::JtoKWH;
+
+    SetPredefinedTables(*state);
+    std::string CompName = "My Coil <coil is DX>";
+
+    PreDefTableEntry(*state, state->dataOutRptPredefined->pdchDXCoolCoilType, CompName, "Coil:Cooling:DX:SingleSpeed");
+    // This would normally be called with numerics such as CompName, 0.006, 8, but I don't really care
+    PreDefTableEntry(*state, state->dataOutRptPredefined->pdch2CoilLvgHumRatIdealPeak, CompName,  "My Design Day where it's >= 8\u00B0");
+    PreDefTableEntry(*state, state->dataOutRptPredefined->pdst2CoilSummaryCoilSelection, CompName,  "My Design Day where it's >= 8\u00B0"); // this is >= 8 degree sign
+
+    // We enable the reports we care about, making sure we have the right ones
+    EXPECT_EQ("HVACSizingSummary", state->dataOutRptPredefined->reportName(6).name);
+    state->dataOutRptPredefined->reportName(6).show = true;
+
+    OutputReportTabular::OpenOutputTabularFile(*state);
+
+    WritePredefinedTables(*state);
+
+    OutputReportTabular::CloseOutputTabularFile(*state);
+
+    std::vector<std::string> lines = read_lines_in_file(DataStringGlobals::outputTblHtmFileName);
```

Comment:
Setup the unit test. We write some stuff to the HTML with special characters. Ensures the HTML is closed. Then we read back the html lines.